src/MiniCover/Instrumentation/Instrumenter.cs

```diff
@@ -213,7 +219,18 @@ private InstrumentedAssembly InstrumentAssemblyIfNecessary(string assemblyFile)
                         ilProcessor.InsertBefore(enterMethodInstruction, pathParamLoadInstruction);
                         UpdateInstructionReferences(methodDefinition, instructions[0], pathParamLoadInstruction);
 
-                        foreach (var instruction in instructions.Values)
+                        //Remove tail call optimization
+                        foreach (var instruction in ilProcessor.Body.Instructions.ToArray())
```

Comment:
I just got familiarized with Tail IL instruction.
That instruction indicates that the next call is the last instruction on that method.
When we add more instructions at the end o the method, we break the IL because Tail doesn't precede the last instruction anymore.
I'm removing all Tail Call optimizations to increase compatibility with F#.
I don't see any issues since .NET also compiles without Tail call optimization when compiling on debug mode.

Reply:
I m almost done with try finally but I have one last problem with lambda and maybe it come from that. so I give a try

Reply:
Why you don't clean Tail after the Return processing?

Reply:
Btw, tail only exists when compiling F#.
You mean like deleting the instruction before ret instruction?
The format is something like:
```
tail
call Void SomeMethod()
ret
```
I think it's easier to do a separate loop and delete all tails instructions.

Reply:
If you mean just switching both for loops, yes, they can be switched. We can process ret before deleting tails.